Dockerfile

```diff
@@ -1,18 +1,24 @@
 # syntax=docker/dockerfile:1
 
-ARG GO_VERSION="1.21.5"
+ARG GO_VERSION="1.22.1"
 ARG OSXCROSS_VERSION="11.3"
+ARG GHQ_VERSION="1.6.1"
 ARG XX_VERSION="1.3.0"
-ARG ALPINE_VERSION="3.18"
+ARG ALPINE_VERSION="3.19"
 ARG PLATFORMS="linux/386 linux/amd64 linux/arm64 linux/arm/v5 linux/arm/v6 linux/arm/v7 linux/mips linux/mipsle linux/mips64 linux/mips64le linux/ppc64le linux/riscv64 linux/s390x windows/386 windows/amd64"
 
 FROM --platform=$BUILDPLATFORM tonistiigi/xx:${XX_VERSION} AS xx
-FROM --platform=$BUILDPLATFORM golang:1.20-alpine${ALPINE_VERSION} AS base
+FROM --platform=$BUILDPLATFORM golang:1.22-alpine${ALPINE_VERSION} AS base
 COPY --from=xx / /
 ENV CGO_ENABLED=0
 RUN apk add --no-cache file git
 WORKDIR /src
 
+FROM base AS ghq
+ARG GHQ_VERSION
+RUN --mount=type=cache,target=/go/pkg/mod \
+  go install github.com/x-motemen/ghq@v${GHQ_VERSION}
```

Comment:
@crazy-max isn't ghq available after this command (my answer is related to your other question) :)